_posts/2020-01-30-get-started-with-java-openjdk11.md

```diff
@@ -0,0 +1,497 @@
+
+```sh
+export DOCKER_BUILDKIT=1
+```
+
+```sh
+faas-cli up -f find-a-quote.yml
+```
+
+My build took around 15s to complete on a modest Intel i7 micro-PC.
+
+You'll be given a URL and Kubernetes will have already pulled the image and started a Pod for your function.
+
+```
+kubectl get deploy -n openfaas-fn -o wide
+```
+
+```sh
+curl -sSLf http://192.168.0.26:31112/function/find-a-quote?q=tree | jq
+```
+
+In the example above I used the `jq` utility to format the output, it looks like there were 10 results for "tree".
+
+```json
+{
+  "count": 18,
+  "results": [
+    {
+      "_id": "5d91b45d9980192a317c8acc",
+      "quoteText": "Notice that the stiffest tree is most easily cracked, while the bamboo or willow survives by bending with the wind.",
+      "quoteAuthor": "Bruce Lee"
+    },
+    {
+      "_id": "5d91b45d9980192a317c8a62",
+      "quoteText": "Notice that the stiffest tree is most easily cracked, while the bamboo or willow survives by bending with the wind.",
+      "quoteAuthor": "Bruce Lee"
+    }
+}
+```
+
+As an extension of this example, why don't you customise the code to return a random index of the quotes found? You'll also want to find yourself a JSON parsing library and then to add it to your `build.gradle` file.
+
+> In my blog post [Java comes to OpenFaaS from 2018](https://blog.alexellis.io/java-comes-to-openfaas/), I used [Gson from Google](https://github.com/google/gson). You'll find an example of how to use the library in that post.
+
+### Example 2) The `java11-vert-x` service
+
+In this example we'll build a service using Vert.x which determine the latest download URL for a GitHub project. We'll set the name and owner of the project via the deployment YAML file, rather than taking it in via the query string.
+
+Why do this? Well downloading binaries from GitHub's releases page is a common task that needs to be automated in many workflows, unfortunately the GitHub API that gives this data back is very heavily rate-limited. Fortunately there's a work-around to send a HTTP `HEAD` request to the GitHub HTTP server instead.
+
+```sh
+faas-cli new --lang java11-vert-x \
+  github-release-finder
+```
+
+Edit `github-release-finder.yml` and add the following for your service:
+
+```yaml
+    environment:
+      owner: openfaas
+      repo: faas-cli
+```
+
+We can read this deployment data at runtime using `System.getenv()`, for confidential data like API tokens, we would use a Kubernetes secret created via `faas-cli secret create` or `kubectl`.
+
+Here's what our basic Handler.java file looks like with the Vert.x template:
+
+```java
+package com.openfaas.function;
+
+import io.vertx.core.http.HttpServerResponse;
+import io.vertx.ext.web.RoutingContext;
+import io.vertx.ext.web.handler.BodyHandler;
+import io.vertx.core.json.JsonObject;
+
+public class Handler implements BodyHandler {
```

Comment:
We can simplify the template to be `Handler<RoutingContext>` instead of `BodyHandler` and in that case you only need to implement this method

Reply:
That sounds better to me, I will try it. Any other changes needed for that?

Reply:
Is that typed as `io.vertx.ext.web.handler.Handler`?